File: 2022-convert-1d-array-into-2d-array/main.py
class Solution:
    def construct2DArray(self, original: List[int], m: int, n: int) -> List[List[int]]:
        
        if len(original) != m * n:
            return []
        
        ans = []

        for i in range(m):
            row = original[i * n: (i + 1) * n]
            ans.append(row)
        
        return ans

File: 2022-convert-1d-array-into-2d-array/main.go
func construct2DArray(original []int, m int, n int) [][]int {
    
    if len(original) != m * n {
        return [][]int{}
    }

    var ans [][]int
    for i := range(m) {
        row := original[i * n:(i + 1) * n]
        ans = append(ans, row)
    }

    return ans
}
